ittapi-rs/ittapi/src/jit.rs

```diff
@@ -0,0 +1,175 @@
+//! The JIT (Just-In-Time) Profiling API provides functionality to report information about
+//! just-in-time generated code that can be used by performance tools. The [Jit] Rust structure is a
+//! high-level view of a subset of the full functionality available. See the [JIT Profiling API] for
+//! more information.
+//!
+//! [JIT Profiling API]:
+///     https://www.intel.com/content/www/us/en/develop/documentation/vtune-help/top/api-support/jit-profiling-api.html
+use anyhow::Context;
+use std::{ffi::CString, os, ptr};
+
+/// Register JIT-compiled methods with a performance tool (e.g., VTune). This structure assumes
+/// single-threaded access; if your program may be multi-threaded, make sure to guard multi-threaded
+/// access with a mutex.
+#[derive(Default)]
+pub struct Jit {
+    did_shutdown: bool,
```

Comment:
This is legacy but as a nit, perhaps we can change this to shutdown_complete in a future patch. Variable name seems not the best.